### PR TITLE
[5.4] Allow passing relative view path to @include

### PR DESCRIPTION
**Pull request by themsaid**
In reference to https://github.com/laravel/internals/issues/183

This PR allows the following:

```
// In view blog/posts/post.blade.php

@include('>sidebar')
```

This is equivalent to:

```
@include('blog.posts.sidebar')
```

**Edit:**
The delimiter was changed to `:`, a class constant is now used to set the delimiter for easy change.

### CODE REVIEW

**Comment by JosephSilber** [comment]
@themsaid curious why you settled on the `>` character. I personally like the `:` more.

**Comment by prateekkathal** [comment]
I agree with @JosephSilber `:` looks much cleaner... A `~` can also be used 😄

File: src/Illuminate/View/Compilers/BladeCompiler.php [hunk 1]
```diff
@@ -951,6 +988,23 @@ public function stripParentheses($expression)
...
     /**
+     * Strip the quotes from the given expression.
+     *
+     * @param  string  $expression
+     * @return string
+     */
+    public function stripQuotes($expression)
+    {
+        foreach (["'", '"'] as $quote) {
+            if (Str::startsWith($expression, $quote)) {
```
Comment (jhoff) [inline src/Illuminate/View/Compilers/BladeCompiler.php]: `startsWith` already accepts an array as the second "needles" argument:

```
if (Str::startsWith($expression, ["'", '"'])) {
    $expression = substr($expression, 1, -1);
}
```

File: src/Illuminate/View/Compilers/BladeCompiler.php [hunk 2]
```diff
@@ -1101,4 +1155,38 @@ public function setEchoFormat($format)
     {
         $this->echoFormat = $format;
     }
+
+    /**
+     * The base of the view currently being compiled.
+     *
+     * @return string
+     */
+    private function viewBase()
```
Comment (jhoff) [inline src/Illuminate/View/Compilers/BladeCompiler.php]: This whole method is kinda hard to grok TBH...

File: src/Illuminate/View/Compilers/BladeCompiler.php [hunk 3]
```diff
@@ -1101,4 +1155,38 @@ public function setEchoFormat($format)
...
+
+    /**
+     * Get the view name from the given expression.
+     *
+     * @param  string  $expression
+     * @return string
+     */
+    public function getViewNameFromExpression($expression)
+    {
+        if ($this->view && Str::startsWith($stripped = $this->stripQuotes($expression), '>')) {
```
Comment (jhoff) [inline src/Illuminate/View/Compilers/BladeCompiler.php]: This could probably be refactored a bit to be more legible.

**Review by jhoff — CHANGES_REQUESTED** [review]
Added a few comments to the code changes. I feel like a different character just doesn't feel right. What if the expression just started with a period? Could that be used as the indicator?

**Comment by themsaid** [comment]
@JosephSilber added the delimiter as a clss constant for easy change since I was sure we might get many other suggestions as well 😃 
https://github.com/laravel/framework/pull/16218/files#diff-ee25397bd547c885a2e12e8894938b8cR19

**Comment by ericbarnes** [comment]
Is it not possible from the coding side to just use the more familiar syntax? 

```
@include('./sidebar')
```

**Comment by JosephSilber** [comment]
Dots and slashes are identical in blade includes, so that would be very confusing.

**Comment by jhoff** [comment]
Why not just a dot?

```
@include('.sidebar')
```

**Comment by JosephSilber** [comment]
A dot would be converted to a leading slash, which usually indicates an absolute path, not a relative one.

**Comment by taylorotwell** [comment]
I'm going to hold off on this for now. I was actually able to implement it just within the Blade compiler itself without any other changes but there is no good way to support it for the `@each` directive and that will cause confusion and be inconsistent.

**Pull request closed without merging**